src/context/zod-options.ts

```diff
@@ -0,0 +1,26 @@
+import { ComponentContext, createContext, useContext } from "@alloy-js/core";
+import { Children } from "@alloy-js/core/jsx-runtime";
+import { Type } from "@typespec/compiler";
+
+export interface ZodOptionsContext {
+  /**
+   * Provide custom component for rendering a specific TypeSpec type.
+   */
+  customTypeComponent: Map<Type, Children>;
```

Comment:
Don't you think there might need to be a more advanced customization where you might need to check a decorator accessor like
```
custom: (type) => {
   if(isError(type)) {
     return <Error/>
  }
   
   return "default";
}
```

Reply:
yeah I can see that